server/lib/friends.js

```diff
@@ -222,6 +223,55 @@ function quitFriends (callback) {
   })
 }
 
+function removeFriend (podId, callback) {
+  // Stop pool requests
+  requestScheduler.deactivate()
+
+  waterfall([
+    function flushRequests (callbackAsync) {
+      requestScheduler.flush(err => callbackAsync(err))
+    },
+
+    function flushVideoQaduRequests (callbackAsync) {
+      requestVideoQaduScheduler.flush(err => callbackAsync(err))
+    },
+
+    function getPod (callbackAsync) {
+      return db.Pod.load(podId, callbackAsync)
```

Comment:
Please create a middleware validator which checks the pod exists and put it in the `req` object (you could take example of videos https://github.com/Chocobozzz/PeerTube/blob/develop/server/middlewares/validators/videos.js#L91).